### ISSUE TITLE
4 years old obsolete code presented as a current version

### ISSUE THREAD

**Issue by nathanaelle** [golang/gddo#415]
hi,

this page is the godoc of a very old version of the "net" package : https://godoc.org/github.com/golang/go/src/pkg/net

there is neiter time nor version indicator that say dont trust this obsolete doc.
the only way to see the misguidance is to read the commit date of the source code used to produce the page.

**Comment by shurcooL** [comment]
This is correct behavior. It might be a little misleading, so let me explain.

If you'd like to see documentation for the `net` package, you should go to its actual import path, which is simply `net`. So:

https://godoc.org/net

What your link points to is something else. The Go source code is mirrored at https://github.com/golang/go. That is not a valid import path, at least it's not designed to be used that way. That repository has a [tag named `go1`](https://github.com/golang/go/tree/go1), and `go get` behavior is to treat that tag with priority over `master` or other branches. From https://golang.org/cmd/go/#hdr-Download_and_install_packages_and_dependencies:

> When checking out or updating a package, get looks for a branch or tag that matches the locally installed version of Go. The most important rule is that if the local installation is running version "go1", get searches for a branch or tag named "go1". If no such version exists it retrieves the most recent version of the package.

So godoc.org correctly follows that behavior.

Closing because it's not an issue in gddo project, but let me know if you have any followup questions.

**Comment by adg** [comment]
We should probably block those paths, though, as they serve no useful
purpose.

**Comment by shurcooL** [comment]
> We should probably block those paths, though, as they serve no useful purpose.

I considered mentioning that, but decided it would be a bad idea.

That would introduce special behavior to the godoc.org which would be inconsistent with the `go get` command. After all, one currently _can_ do `go get -d github.com/golang/go/src/pkg/net` and it will do what's expected based on the [rules](https://golang.org/cmd/go/#hdr-Remote_import_paths) for how `go get` works. I don't think godoc.org should try to deviate from that, it's not its responsibility.

However, what _might_ be a good idea is to display a yellow banner warning on top, something like:

> If you're looking for documentation for Go standard library, go [here](https://godoc.org/-/go) instead.

Similar to how godoc redirects and displays a yellow banner for when you visit a Go import path that doesn't match the canonical one (as specified in an import comment):

![image](https://cloud.githubusercontent.com/assets/1924134/15808773/352db8b0-2b35-11e6-8a8c-3575604f91a9.png)

(Except, we probably shouldn't automatically redirect; or at least make it possible to view that weird but valid import path.)

**Comment by shurcooL** [comment]
> We should probably block those paths, though, as they serve no useful purpose.

Then again, if there's already code in gddo to block certain paths, and adding these is easy, then perhaps that's the best thing to do. I agree that they serve no useful purpose, and someone really determined can just use local `godoc` command or something.

**Comment by adg** [comment]
There is a mechanism to block certain paths, and I think it would be a usability improvement to block these paths.

**Comment by shantuo** [comment]
I'm working on removing obsolete packages which are inactive for long time and with no imports periodically.

I wonder where the rule that gddo tries to look for go1 tag still affects. Instead, our current logic when [getting a GitHub directory](https://github.com/golang/gddo/blob/master/gosrc/github.go#L71) in most cases falls back to getting master branches, which is complained in #405 as well. I think it's time to change it to get the default branch.
